infoblox_client/object_manager.py

```diff
@@ -161,16 +161,21 @@ def update_network_options(self, ib_network, extattrs=None):
                 ib_network.extattrs = extattrs
         return ib_network.update()
 
-    def get_host_record(self, dns_view, ip):
-        return obj.HostRecord.search(self.connector,
-                                     view=dns_view,
-                                     ip=ip)
-
-    def find_hostname(self, dns_view, hostname, ip):
-        return obj.HostRecord.search(self.connector,
-                                     name=hostname,
-                                     view=dns_view,
-                                     ip=ip)
+    def get_host_record(self, dns_view, ip, network_view=None):
+        kwargs = {'ip': ip}
```

Comment:
This processing can be simplified.
Base object always excludes values that are None from search request, so all you need here is just add one more parameter to HostRecord search (network_view=network_view).

[1] https://github.com/infobloxopen/infoblox-client/blob/master/infoblox_client/objects.py#L100